File: Python programs/prime_number.py
def prime(n):
    if n<=1:
        return False
    if n==2:
        return True
    if n%2==0:
        return False
    for i in range(3,int(n**0.5)+1,2):
        if n%i==0:
            return False
    return True
number = int(input("Enter the number: "))
if prime(number):
    print(f"{number} is prime number.")
else:
    print(f"{number} is not a prime number.")

File: CPP_programs/prime_number.cpp
#include<iostream>
#include<cmath>
using namespace std;
bool is_prime(int n);
int main(){
    int n;
    cout<<"Enter the number: ";
    cin>>n;
    if (is_prime(n)){
        cout<<n<<" is a prime number"<<endl;
    }
    else{
        cout<<n<<" is not a prime number"<<endl;
    }
    return 0;
}
bool is_prime(int n){
    if (n<=1){
        return 0;
    }
    if (n==2){
        return 1;
    }
    if(n%2==0){
        return 0;
    }
    for (int i=3; i<=sqrt(n);i+=2){
        if(n%i==0){
            return 0;
        }
    }
    return 1;
}